### ISSUE TITLE
Hitting API request limits

### ISSUE THREAD

**Issue by pmsumner** [Graylog2/graylog-plugin-aws#14]
I've set up reading flowlogs into Graylog for our dev VPC to test out and see what information we can get from them.

I've almost immediately hit a problem where a number of requests get 400 responses from AWS:

```
2016-09-07_07:31:26.66398 2016-09-07 08:31:26,663 ERROR: org.graylog.aws.inputs.flowlogs.FlowLogReader - Could not read AWS FlowLogs from stream [eni-b8******-reject].
2016-09-07_07:31:26.66440 com.amazonaws.AmazonServiceException: Rate exceeded (Service: AWSLogs; Status Code: 400; Error Code: ThrottlingException; Request ID: 1ebbcb03-74cd-11e6-8780-9bffc48ed044)
2016-09-07_07:31:26.66496       at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1377) ~[graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.66611       at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:923) ~[graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.66660       at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:701) ~[graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.66744       at com.amazonaws.http.AmazonHttpClient.doExecute(AmazonHttpClient.java:453) ~[graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.66787       at com.amazonaws.http.AmazonHttpClient.executeWithTimer(AmazonHttpClient.java:415) ~[graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.66875       at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:364) ~[graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.66921       at com.amazonaws.services.logs.AWSLogsClient.doInvoke(AWSLogsClient.java:1962) ~[graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.67057       at com.amazonaws.services.logs.AWSLogsClient.invoke(AWSLogsClient.java:1932) ~[graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.67096       at com.amazonaws.services.logs.AWSLogsClient.getLogEvents(AWSLogsClient.java:1431) ~[graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.67204       at org.graylog.aws.inputs.flowlogs.FlowLogReader.readStream(FlowLogReader.java:153) [graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.67253       at org.graylog.aws.inputs.flowlogs.FlowLogReader.run(FlowLogReader.java:122) [graylog-plugin-aws-1.0.0.jar:?]
2016-09-07_07:31:26.67346       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_101]
2016-09-07_07:31:26.67388       at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_101]
2016-09-07_07:31:26.67472       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_101]
2016-09-07_07:31:26.67521       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_101]
2016-09-07_07:31:26.67658       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_101]
2016-09-07_07:31:26.67715       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_101]
2016-09-07_07:31:26.67804       at java.lang.Thread.run(Thread.java:745) [?:1.8.0_101]
```

Originally I was seeing many of these a second, but then I disabled the IP->name lookups and the volume has dropped to several a minute.

I asked AWS support if they'd increase the limit for GetLogEvents calls and their response was "yes, but..."

> The solution is to use the Subscriptions facility, which turns the egress from the Logs service from a polling interface to a "push" interface that scales

**Comment by pmsumner** [comment]
Input setup:
`
aws_region: eu-west-1
log_group_name: flowLogs
max_backtrack: 12
read_delay_minutes: 15
throttling_allowed: false
`


`Instance detail lookups: Disabled
Lookup regions: eu-west-1,eu-central-1
Access Key: ***********
Secret Key: ***********
`

**Comment by skny5** [comment]
Facing the same issue, any recommendation?

**Comment by lennartkoopmann** [comment]
I'm working on a new implementation that uses Kinesis Streams instead of polling the APIs.

**Comment by lennartkoopmann** [comment]
Implemented in f0f057c08f0ffc310d3dfa67970ff6b9472efb0c.

Now working on documentation and will issue a new release soon.

**Comment by lennartkoopmann** [comment]
Released in v1.2.0: https://github.com/Graylog2/graylog-plugin-aws/releases/tag/1.2.0

Please upgrade and re-open this issue if there are any problems. Also please let me know if there are any problems with the documentation.

Thanks! :)
